llvm/lib/Target/DirectX/DXILPostOptimizationValidation.cpp

```diff
@@ -24,6 +26,48 @@ using namespace llvm;
 using namespace llvm::dxil;
 
 namespace {
+static const char *ResourceClassToString(llvm::dxil::ResourceClass Class) {
+  switch (Class) {
+  case ResourceClass::SRV:
+    return "SRV";
+  case ResourceClass::UAV:
+    return "UAV";
+  case ResourceClass::CBuffer:
+    return "CBuffer";
+  case ResourceClass::Sampler:
+    return "Sampler";
+  }
```

Comment:
There is an exactly equivalent function to this in the DXILResource.cpp file. We should refactor and have a single implementation instead of each file getting its own copy of this function.